### PR TITLE
Refresh the scipy chapter

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
This pull request merges in the changes that I did for the scipy tutorial at EuroScipy 2017.

Changes can be previewed on http://gael-varoquaux.info/scipy-lecture-notes/intro/scipy.html

A brief over of the changes:

- Minor CSS changes for better rendering
- Rewrite of the scipy intro chapter:
  - Actual content was most left unchanged, mostly wording and style
  - Change code to use more modern scipy idioms
  - Reorder the chapter to be more didactic
  - Use sphinx-gallery in many place to have more didactic examples
  - Add seealso and links
  - Bump the scipy version to 0.16.1 (more recent, but not bleeding edge).

This PR comprises the changes in #287

### CODE REVIEW

File: intro/image_processing/image_processing.rst [hunk 1]
```diff
@@ -226,39 +237,63 @@ structuring element centered on the pixel of interest. ::
            [0, 0, 0, 0, 0, 0, 0]])
 
 
-Measurements on images
-........................
+Connect components and measurements on images
```
Comment (emmanuelle) [inline intro/image_processing/image_processing.rst]: Connect or connected?

Comment (GaelVaroquaux) [reply intro/image_processing/image_processing.rst]: Fixed

File: intro/image_processing/image_processing.rst [hunk 2]
```diff
@@ -226,39 +237,63 @@ structuring element centered on the pixel of interest. ::
...
+    :target: scipy/auto_examples/plot_connect_measurements.html
+    :scale: 60
+    :align: center
+
+.. image:: scipy/auto_examples/images/sphx_glr_plot_connect_measurements_002.png
+    :target: scipy/auto_examples/plot_connect_measurements.html
+    :scale: 60
+    :align: right
+
+:func:`scipy.ndimage.label` assigns a different label to each connect
```
Comment (emmanuelle) [inline intro/image_processing/image_processing.rst]: here too, connected

File: intro/image_processing/image_processing.rst [hunk 3]
```diff
@@ -226,39 +237,63 @@ structuring element centered on the pixel of interest. ::
...
     >>> labels, nb = ndimage.label(mask)
     >>> nb
     8
+
+.. raw:: html
+
+   <div style="clear: both"></div>
+
+
+Now compute measurements on each connect component::
```
Comment (emmanuelle) [inline intro/image_processing/image_processing.rst]: here too

File: intro/image_processing/image_processing.rst [hunk 4]
```diff
@@ -226,39 +237,63 @@ structuring element centered on the pixel of interest. ::
...
+
     >>> areas = ndimage.sum(mask, labels, range(1, labels.max()+1))
-    >>> areas
+    >>> areas   # The number of pixels in each connect component
```
Comment (emmanuelle) [inline intro/image_processing/image_processing.rst]: connected

File: intro/image_processing/image_processing.rst [hunk 5]
```diff
@@ -226,39 +237,63 @@ structuring element centered on the pixel of interest. ::
...
     array([ 190.,   45.,  424.,  278.,  459.,  190.,  549.,  424.])
     >>> maxima = ndimage.maximum(sig, labels, range(1, labels.max()+1))
-    >>> maxima
+    >>> maxima  # The maximum signal in each connect component
```
Comment (emmanuelle) [inline intro/image_processing/image_processing.rst]: connected

File: intro/image_processing/image_processing.rst [hunk 6]
```diff
@@ -226,39 +237,63 @@ structuring element centered on the pixel of interest. ::
...
     array([  1.80238238,   1.13527605,   5.51954079,   2.49611818,
              6.71673619,   1.80238238,  16.76547217,   5.51954079])
+
+.. image:: scipy/auto_examples/images/sphx_glr_plot_connect_measurements_003.png
+    :target: scipy/auto_examples/plot_connect_measurements.html
+    :scale: 60
+    :align: right
+
+
+Extract the 4th connect component, and crop the array around it::
```
Comment (emmanuelle) [inline intro/image_processing/image_processing.rst]: connected

File: intro/scipy/examples/plot_connect_measurements.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
+"""
+=============================
+Demo connect components
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_connect_measurements.py]: connected

File: intro/scipy/examples/plot_connect_measurements.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+=============================
+
+Extracting and labeling connect components in a 2D array
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_connect_measurements.py]: connected

**Comment by GaelVaroquaux** [comment]
Addressed all the "connected" -> "connected"

File: intro/scipy/examples/plot_connect_measurements.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+plt.subplot(1, 2, 2)
+plt.imshow(mask, cmap=plt.cm.gray)
+plt.axis('off')
+plt.title('mask')
+plt.subplots_adjust(wspace=.05, left=.01, bottom=.01, right=.99, top=.9)
+
+
+############################################################
+# Label connect components
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_connect_measurements.py]: connected

File: intro/scipy/examples/solutions/plot_fft_image_denoise.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+############################################################
+# Compute the 2d FFT of the input image
+############################################################
+from scipy import fftpack
+im_fft = fftpack.fft2(im)
+
+# Show the results
+
+def plot_spectrum(im_fft):
```
Comment (emmanuelle) [inline intro/scipy/examples/solutions/plot_fft_image_denoise.py]:
```
    from matplotlib.colors import LogNorm
    plt.imshow(np.abs(im_fft), norm=LogNorm(vmin=5))
```

File: intro/scipy/examples/solutions/plot_fft_image_denoise.py [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+# Call ff a copy of the original transform. Numpy arrays have a copy
+# method for this purpose.
+im_fft2 = im_fft.copy()
+
+# Set r and c to be the number of rows and columns of the array.
+r, c = im_fft2.shape
+
+# Set to zero all rows with indices between r*keep_fraction and
+# r*(1-keep_fraction):
+im_fft2[r*keep_fraction:r*(1-keep_fraction)] = 0
```
Comment (emmanuelle) [inline intro/scipy/examples/solutions/plot_fft_image_denoise.py]: I get the following error (numpy 1.13.1)
```
    TypeError: slice indices must be integers or None or have an __index__ method
```

Comment (emmanuelle) [inline intro/scipy/examples/solutions/plot_fft_image_denoise.py]: Maybe explain how coefficients are organized?

File: intro/scipy/examples/plot_t_test.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+==========================================
+Comparing 2 set of samples from Gaussians
+==========================================
+
+"""
+
+import numpy as np
+from matplotlib import pyplot as plt
+
+# Generates 2 set of observations
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_t_test.py]: 2 sets

File: intro/scipy/examples/plot_t_test.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
+"""
+==========================================
+Comparing 2 set of samples from Gaussians
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_t_test.py]: 2 sets

File: intro/scipy/examples/plot_t_test.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+# Generates 2 set of observations
+samples1 = np.random.normal(0, size=1000)
+samples2 = np.random.normal(1, size=1000)
+
+# Compute a histogram of the sample
+bins = np.linspace(-4, 4, 30)
+histogram1, bins = np.histogram(samples1, bins=bins, normed=True)
+histogram2, bins = np.histogram(samples2, bins=bins, normed=True)
+
+bin_centers = 0.5*(bins[1:] + bins[:-1])
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_t_test.py]: is ``bin_centers`` needed?

File: intro/scipy/examples/plot_image_filters.py [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+plt.imshow(median_face, cmap=plt.cm.gray)
+plt.axis('off')
+plt.title('median filter')
+
+plt.subplot(144)
+plt.imshow(wiener_face, cmap=plt.cm.gray)
+plt.title('Wiener filter')
+plt.axis('off')
+
+plt.subplots_adjust(wspace=.05, left=.01, bottom=.01, right=.99, top=.99)
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_image_filters.py]: ``plt.tight_layout()`` ?

File: intro/scipy/examples/plot_image_transform.py [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+plt.subplot(154)
+plt.imshow(cropped_face, cmap=plt.cm.gray)
+plt.axis('off')
+
+plt.subplot(155)
+plt.imshow(zoomed_face, cmap=plt.cm.gray)
+plt.axis('off')
+
+plt.subplots_adjust(wspace=.05, left=.01, bottom=.01, right=.99, top=.99)
```
Comment (emmanuelle) [inline intro/scipy/examples/plot_image_transform.py]: ``plt.tight_layout()`` ?

**Comment by emmanuelle** [comment]
A few comments, otherwise LGTM

**Comment by GaelVaroquaux** [comment]
> plt.tight_layout() ?

Doesn't adjust the wspace.

**Comment by GaelVaroquaux** [comment]
> plt.tight_layout() ?

Setting the horizontal space between subplots is very important here.

**Comment by emmanuelle** [comment]
Merging.

**Pull request merged**